cristin-organization/src/main/java/no/unit/nva/cristin/organization/HttpExecutorImpl.java

```diff
@@ -96,20 +98,47 @@ protected SubSubUnitDto fetch(URI uri) throws InterruptedException, NotFoundExce
         }
     }
 
+    protected SearchResponse<Organization> query(URI uri) throws NotFoundException, FailedHttpRequestException {
+        HttpResponse<String> response = sendRequest(createHttpRequest(addLanguage(uri)));
+        if (isSuccessful(response.statusCode())) {
+            try {
+                List<Organization> organizations = getOrganizations(response);
+                return new SearchResponse<Organization>(uri)
+                        .withHits(organizations)
+                        .withSize(getCount(response, organizations));
+            } catch (JsonProcessingException e) {
+                throw new FailedHttpRequestException(e.getMessage());
+            }
+        } else {
+            throw new NotFoundException(String.format(NOT_FOUND_MESSAGE_TEMPLATE, uri));
+        }
+    }
+
+    private int getCount(HttpResponse<String> response, List<Organization> organizations) {
+        final int count = response.headers().firstValue("X-Total-Count").isPresent()
+                ? Integer.parseInt(response.headers().firstValue("X-Total-Count").get())
+                : organizations.size();
+        return count;
+    }
+
+    private List<Organization> getOrganizations(HttpResponse<String> response) throws JsonProcessingException {
+        List<SubUnitDto> units = OBJECT_MAPPER.readValue(response.body(), new TypeReference<List<SubUnitDto>>() {
+        });
+        return units.stream()
+                .map(SubUnitDto::getUri)
+                .map(uri -> attempt(() -> getOrganization(uri)).orElseThrow())
+                .collect(Collectors.toList());
+    }
+
     private HttpRequest createHttpRequest(URI uri) {
         return HttpRequest.newBuilder()
                 .GET()
                 .uri(uri)
                 .build();
     }
 
-    private HttpResponse<String> sendRequest(HttpRequest httpRequest) throws InterruptedException,
-            FailedHttpRequestException {
-        try {
-            return httpClient.sendAsync(httpRequest, BodyHandlers.ofString()).get();
-        } catch (ExecutionException e) {
-            throw new FailedHttpRequestException(e);
-        }
+    protected HttpResponse<String> sendRequest(HttpRequest httpRequest) {
+        return attempt(() -> httpClient.sendAsync(httpRequest, BodyHandlers.ofString()).get()).orElseThrow();
```

Comment:
Not sure if you could use some parts from ApiClient in cristin-commons in this class